client/components/Dropdown.jsx

```diff
@@ -1,25 +1,88 @@
+import {createRef} from 'preact';
 import PureComponent from '../lib/PureComponent';
 
 import s from './Dropdown.css';
+
 export default class Dropdown extends PureComponent {
+  input = createRef();
+
+  state = {
+    query: '',
+    showOptions: false
+  };
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClickOutside, true);
```

Comment:
couldn't use `onBlur` since clicking an option would hide the list before the click event was fired